src/main/java/io/cdap/plugin/ariba/source/AribaInputFormat.java

```diff
@@ -0,0 +1,139 @@
+    AribaPluginConfig pluginConfig = getPluginConfig(jobContext);
+    AribaServices aribaServices = new AribaServices(pluginConfig);
+    boolean previewEnabled = Boolean.parseBoolean(jobContext.getConfiguration().
+                                                    get(ResourceConstants.IS_PREVIEW_ENABLED));
+    createJob(pluginConfig, aribaServices, previewEnabled, null);
+    return resultSplits;
+  }
+
+  @Override
+  public RecordReader<NullWritable, StructuredRecord> createRecordReader(InputSplit inputSplit, TaskAttemptContext
+    taskAttemptContext) throws IOException {
+    AribaPluginConfig pluginConfig = getPluginConfig(taskAttemptContext);
+    AribaServices aribaServices = new AribaServices(pluginConfig);
+    Schema outputSchema = Schema.parseJson(taskAttemptContext.getConfiguration().get(ResourceConstants.OUTPUT_SCHEMA));
+    return new AribaRecordReader(aribaServices, outputSchema);
+  }
+
+  private AribaPluginConfig getPluginConfig(JobContext taskAttemptContext) {
+    return GSON.fromJson(taskAttemptContext.getConfiguration().
+                           get(ResourceConstants.ARIBA_PLUGIN_PROPERTIES), AribaPluginConfig.class);
+  }
+
+  /**
+   *  call for create Job
+   *
+   * @param pluginConfig  Plugin config
+   * @param aribaServices  Ariba services
+   * @param isPreviewEnabled Is preview enabled
+   * @param pageToken    Page token
+   * @throws IOException
+   */
+  private void createJob(AribaPluginConfig pluginConfig,
+                         AribaServices aribaServices, boolean isPreviewEnabled,
+                         String pageToken) throws IOException {
+    JsonNode jobData;
+    String jobId;
+
+    try {
+      JsonNode createJobResponse = aribaServices.createJob(pluginConfig,
+                                                           aribaServices.getAccessToken(), pageToken);
+      jobId = createJobResponse.get(ResourceConstants.JOB_ID).asText();
+    } catch (AribaException | InterruptedException | IOException exception) {
+      throw new IOException(exception.getMessage(), exception);
+    }
+
+    while (true) {
+      try {
+        LOG.info("Fetching Data For Job Id: {}", jobId);
+        jobData = aribaServices.fetchJobStatus(aribaServices.getAccessToken(), jobId);
+        if (jobData.get(ResourceConstants.STATUS).asText().equals(ResourceConstants.COMPLETED) ||
+          jobData.get(ResourceConstants.STATUS).asText().equals(ResourceConstants.COMPLETED_ZERO_RECORDS)) {
+          break;
+        }
+        Thread.sleep(60000);
+      } catch (AribaException exception) {
+        if (ResourceConstants.LIMIT_EXCEED_ERROR_CODE.equals(exception.getErrorCode())) {
+          LOG.info("Limit Reached For Token: {}", pageToken);
+          return;
+        } else {
+          throw new IOException(exception.getMessage(), exception);
+        }
+      } catch (InterruptedException | IOException exception) {
+        throw new IOException(exception.getMessage(), exception);
```

Comment:
no need to catch IOException and rethrow it

Reply:
Fixed

Reply:
where is this fixed?

Reply:
This is fixed under this commit(https://github.com/data-integrations/ariba-plugins/pull/2/commits/4c63631ded44b52120d0ad7a20c370dc4ebd5f2b)